addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -586,6 +586,11 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
         if (!this.ribbons[ribbonId]) {
             $editableDocument.find(`[data-ribbon-id="${ribbonId}"]`).each((index, product) => delete product.dataset.ribbonId);
         }
+
+        const $editable = this.$ribbon.parent().find('.o_editable');
+        if ($editable.length) {
+            $editable.addClass('o_dirty');
+        }
```

Comment:
> we will mark dirty even while previewing (I think it's not big deal)

Probably not a big deal indeed. Every preview at the moment marks the DOM as dirty. This is something I wanna change/fix at some point but... not a big deal at the moment.

However, this is probably not the right fix: we normally never have to add that dirty flag by ourselves. If the editable section has savable content, it is dirty if it changed... and if its DOM changed, the `o_dirty` class should have been added automatically. If that's not the case, this is what should be fixed (or explained with code comments why this has to be an exception).

Note: I don't think this is the first time this has been fixed, a proper investigation should be made once and for all.